app/connectors/ChargeHistoryDetailsConnector.scala

```diff
@@ -18,31 +18,28 @@ package connectors
 
 import config.MicroserviceAppConfig
 import connectors.httpParsers.ChargeHistoryHttpParser.{ChargeHistoryReads, ChargeHistoryResponse}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ChargeHistoryDetailsConnector @Inject()(val http: HttpClient,
+class ChargeHistoryDetailsConnector @Inject()(val http: HttpClientV2,
                                               val appConfig: MicroserviceAppConfig
                                              )(implicit ec: ExecutionContext) extends RawResponseReads {
 
 
-  def listChargeHistoryDetailsUrl(idType: String, idNumber: String, regimeType: String): String =
-    s"${appConfig.ifUrl}/cross-regime/charges/$idType/$idNumber/$regimeType"
+  def listChargeHistoryDetailsUrl(idType: String, idNumber: String, regimeType: String, chargeReference: String): String =
```

Comment:
can we make this method private? as its used only in the scope of this connector